packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/auth/ApiKeyAuth.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.mongodb.auth
+
+import io.realm.kotlin.mongodb.App
+import io.realm.kotlin.mongodb.User
+import io.realm.kotlin.mongodb.exceptions.AppException
+import io.realm.kotlin.types.ObjectId
+
+/**
+ * Exposes functionality for a user to manage API keys under their control.
+ */
+public interface ApiKeyAuth {
+
+    /**
+     * The User that this instance in associated with.
+     */
+    public val user: User
+
+    /**
+     * The App that this instance in associated with.
+     */
+    public val app: App
+
+    /**
+     * Creates a user API key that can be used to authenticate as the user.
```

Comment:
These documentation blocks can improve formatting take a look into this example:
https://github.com/realm/realm-kotlin/blob/760055a5bfa1df1c1bb98eb24335772fcd198e74/packages/library-base/src/commonMain/kotlin/io/realm/kotlin/query/RealmResults.kt#L38

Separate paragraphs, @throws to the end